IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -96,4 +96,130 @@ + (nullable NSDictionary *)getRegisteredDeviceMetadataInformation:(nullable id<M
     return nil;
 }
 
++ (nullable MSIDWPJKeyPairWithCert *)findWPJRegistrationInfoWithAdditionalPrivateKeyAttributes:(nonnull NSDictionary *)queryAttributes
+                                                                                certAttributes:(nullable NSDictionary *)certAttributes
+                                                                                       context:(nullable id<MSIDRequestContext>)context
+{
+    OSStatus status = noErr;
+    CFTypeRef privateKeyCFDict = NULL;
+    
+    // Set the private key query dictionary.
+    NSMutableDictionary *queryPrivateKey = [NSMutableDictionary new];
```

Comment:
Thanks, for now I've added additional parameter to hardcode to RSA such as our future code wouldn't accidentally find an ECC key and know nothing about what to do about it (since I'm expecting multiple PkeyAuth to be released earlier than ECC PkeyAuth given server side deployment status). 

`queryPrivateKey[(__bridge id)kSecAttrKeyType] = (__bridge id)kSecAttrKeyTypeRSA;`

Once your other ECC changes are ready, I think you can just remove the extra RSA line I've added, and it should either find ECC or RSA key. We shouldn't have both ECC and RSA key in keychain with the same tag/tenantId, so it should uniquely match it, and your subsequent code will be able to detect if it's ECC or RSA. Let me know if this would work.

Reply:
Sounds good